neo/Consensus/Helper.cs

```diff
@@ -19,6 +19,10 @@ internal static class Helper
         public static bool WatchOnly(this IConsensusContext context) => context.MyIndex < 0;
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static Header PrevHeader(this IConsensusContext context) => context.Snapshot.GetHeader(context.PrevHash);
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static int CountCommitted(this IConsensusContext context) => context.CommitPayloads.Count(p => p != null);
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static int CountFailed(this IConsensusContext context) => context.LastSeenMessage.Count(p => p < (((int) context.BlockIndex) - 1));
```

Comment:
How about if timestamp is older than 5 minutes here we can also consider failed? I expect we can add another timer also requesting recovery every block time When appropriate. Then even the edge cases will be fixed for a node dieing where it could stall.

Reply:
We can make this change in a subsequent PR.